src/OidcClient.php

```diff
@@ -44,13 +49,18 @@ public function __construct(
       private string $clientSecret,
       private string $redirectRoute,
       private string $rememberMeParameter,
-      protected ?OidcWellKnownParserInterface $wellKnownParser = null)
-  {
+      private ?string $codeChallengeMethod,
+      protected ?OidcWellKnownParserInterface $wellKnownParser = null,
+  ) {
     // Check for required phpseclib classes
     if (!class_exists('\phpseclib\Crypt\RSA') && !class_exists(RSA::class)) {
       throw new RuntimeException('Unable to find phpseclib Crypt/RSA.php.  Ensure phpseclib/phpseclib is installed.');
     }
 
+    if ($codeChallengeMethod && !in_array($codeChallengeMethod, $this->getCodeChallengeMethodsSupported(), true)) {
```

Comment:
I do not think the constructor is the right place for this check unfortunately, as it will trigger a configuration load on each construct. Let move it to within the `authenticate` call, after checking whether the code challenge is enabled.

Reply:
What we should do here though, is check whether the `$codeChallengeMode` is supported by this class, which is currently done in the generate challenge method. It can throw a LogicException when that fails.